kolibri/core/assets/src/vue/content-renderer/index.vue

```diff
@@ -81,55 +79,39 @@
-    },
     created() {
-      this.findRendererComponent();
+      this.getRendererComponent();
       // This means this component has to be torn down on channel switches.
-      this.$watch('files', this.findRendererComponent);
+      this.$watch('files', this.getRendererComponent);
     },
     mounted() {
       this.ready = true;
       this.renderContent();
     },
     data: () => ({
       currentViewClass: null,
+      noRendererAvailable: false,
     }),
     methods: {
       /**
-       * Clear any active listeners - this ensures that if the current content node changes,
-       * but the previous one has not received its renderer callback, no shenanigans will occur.
-       */
-      clearListeners() {
-        this._eventListeners.forEach((listener) => {
-          this.Kolibri.off(listener.event, listener.callback);
-        });
-        this._eventListeners = [];
-      },
-      /**
-       * Broadcast through the Kolibri core app for a content renderer module that is able to
+       * Check the Kolibri core app for a content renderer module that is able to
        * handle the rendering of the current content node. This is the entrance point for changes
        * in the props,so any change in the props will trigger this function first.
        */
-      findRendererComponent() {
-        // Clear any existing listeners so that two renderings do not collide in this component.
-        this.clearListeners();
-        // Only bother to do this is if the node is available. Otherwise the template can handle it.
-        if (this.available) {
+      getRendererComponent() {
+        // Assume we will find a renderer until we find out otherwise.
+        this.noRendererAvailable = false;
+        // Only bother to do this is if the node is available, and the kind and extension are defined.
+        // Otherwise the template can handle it.
+        if (this.available && this.kind && this.extension) {
           // The internal content rendering component is currently unrendered.
           this.rendered = false;
-          // This is the event that content renderers will broadcast in response to our call.
-          const event = `component_render:${this.contentType}`;
-          // This is the method that will accept the component passed by the content renderer.
-          const callback = this.setRendererComponent;
-          // Set up listening for the response.
-          this.Kolibri.once(event, callback);
-          // Keep a track of this listener so that we can unbind it later if needed.
-          this._eventListeners.push({ event, callback });
-          // This is the event that is broadcast out to the content renderers.
-          this.Kolibri.emit(`content_render:${this.contentType}`);
-          logging.debug(`Looking for content renderer for ${this.contentType}`);
+          this.Kolibri.retrieveContentRenderer(this.kind, this.extension).then((component) => {
+            this.setRendererComponent(component);
```

Comment:
might be nice to move this function inline here since it's only ever used as a callback